casper/src/main/scala/coop/rchain/casper/engine/BlockApproverProtocol.scala

```diff
@@ -178,32 +179,51 @@ object BlockApproverProtocol {
         _ <- (blockDeploys.size == genesisBlessedContracts.size)
               .either(())
               .or("Mismatch between number of candidate deploys and expected number of deploys.")
-      } yield (blockDeploys, block.body.state)
```

Comment:
The bug was here, what had been validated were processed deploys from unapproved block. So it does not matter what you put in local config, validation only was insuring that unapproved block received is replayable against its own processed deploys.